### PR TITLE
Deprecate promo codes

### PR DESCRIPTION
**Pull request by danwalkowski**
@aaustin @derrickstaten

### CODE REVIEW

**Comment by danwalkowski** [comment]
I wasn't expecting failures from deprecating methods. (?)

File: Branch-SDK/Branch-SDK/Branch.h [hunk 1]
```diff
@@ -112,15 +112,15 @@ typedef NS_ENUM(NSUInteger, BranchCreditHistoryOrder) {
     BranchLeastRecentFirst
 };
 
-typedef NS_ENUM(NSUInteger, BranchPromoCodeRewardLocation) {
-    BranchPromoCodeRewardReferredUser = 0,
-    BranchPromoCodeRewardReferringUser = 2,
-    BranchPromoCodeRewardBothUsers = 3
+typedef NS_ENUM(NSUInteger, BranchPromoCodeRewardLocation ) {
```
Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/Branch.h]: let's remove the space here

File: Branch-TestBed/Podfile.lock [hunk 2]
```diff
@@ -7,4 +7,4 @@ DEPENDENCIES:
 SPEC CHECKSUMS:
   OCMock: a10ea9f0a6e921651f96f78b6faee95ebc813b92
 
-COCOAPODS: 0.37.2
+COCOAPODS: 0.39.0
```
Comment (derrickstaten) [inline Branch-TestBed/Podfile.lock]: In the future, let's keep this in a separate PR / diff. same for changes to `Branch-TestBed/Branch-TestBed.xcodeproj/project.pbxproj` above

**Comment by erickreutz** [comment]
Hey guys. Are you planning to drop support for promo codes in the future? I can't find anywhere but here where you guys state that you are deprecating this feature. Seems like a pretty core function of branch to not mention the deprecation of it publicly anywhere besides this PR?

**Comment by aaustin** [comment]
Hey @erickreutz- I'm really sorry about this. We built this as a hackathon project about a year ago and released it for fun. People liked it a lot and wanted us to build more on top of it. We were spending a ton of time supporting and scaling the service, which was detracting from our focus on deep links. Due to resource constraints, we had to make the call to stop improving and scaling it. 

We won't remove it, but you'll notice that we removed all documentation around it so that no new users will adopt it. The deprecation warnings serve to let you know that we won't be improving that portion of the service. We did one last retrofit a few months ago that should support the current scale indefinitely.

Separately, if you want to build a SAAS company offering referral codes, we'll promise to drive you a ton of users :)

**Pull request merged**